src/ripple/core/SQLInterface.h

```diff
@@ -0,0 +1,224 @@
+
+    virtual std::optional<LedgerIndex>
+    getMaxLedgerSeq() = 0;
+
+    virtual void
+    deleteTransactionByLedgerSeq(LedgerIndex ledgerSeq) = 0;
+
+    virtual void
+    deleteBeforeLedgerSeq(LedgerIndex ledgerSeq) = 0;
+
+    virtual void
+    deleteTransactionsBeforeLedgerSeq(LedgerIndex ledgerSeq) = 0;
+
+    virtual void
+    deleteAccountTransactionsBeforeLedgerSeq(LedgerIndex ledgerSeq) = 0;
+
+    virtual int
+    getTransactionCount() = 0;
+
+    virtual int
+    getAccountTransactionCount() = 0;
+
+    virtual struct CountMinMax
+    getLedgerCountMinMax() = 0;
+
+    virtual bool
+    saveValidatedLedger(
+        std::shared_ptr<Ledger const> const& ledger,
+        bool current) = 0;
+
+    virtual std::optional<LedgerInfo>
+    loadLedgerInfoByIndex(LedgerIndex ledgerSeq) = 0;
+
+    virtual std::optional<LedgerInfo>
+    loadNewestLedgerInfo() = 0;
+
+    virtual std::optional<LedgerInfo>
+    loadLimitedOldestLedgerInfo(LedgerIndex ledgerFirstIndex) = 0;
+
+    virtual std::optional<LedgerInfo>
+    loadLimitedNewestLedgerInfo(LedgerIndex ledgerFirstIndex) = 0;
+
+    virtual std::optional<LedgerInfo>
+    loadLedgerInfoByHash(uint256 const& ledgerHash) = 0;
+
+    virtual uint256
+    getHashByIndex(LedgerIndex ledgerIndex) = 0;
+
+    virtual std::optional<LedgerHashPair>
+    getHashesByIndex(LedgerIndex ledgerIndex) = 0;
+
+    virtual std::map<LedgerIndex, LedgerHashPair>
+    getHashesByIndex(LedgerIndex minSeq, LedgerIndex maxSeq) = 0;
+
+    virtual std::vector<std::shared_ptr<Transaction>>
+    loadTxHistory(LedgerIndex startIndex) = 0;
+
+    virtual AccountTxs
+    getOldestAccountTxs(AccountTxOptions const& options) = 0;
+
+    virtual AccountTxs
+    getNewestAccountTxs(AccountTxOptions const& options) = 0;
+
+    virtual MetaTxsList
+    getOldestAccountTxsB(AccountTxOptions const& options) = 0;
+
+    virtual MetaTxsList
+    getNewestAccountTxsB(AccountTxOptions const& options) = 0;
+
+    virtual std::pair<AccountTxs, std::optional<AccountTxMarker>>
+    oldestAccountTxPage(AccountTxPageOptions const& options) = 0;
+
+    virtual std::pair<AccountTxs, std::optional<AccountTxMarker>>
+    newestAccountTxPage(AccountTxPageOptions const& options) = 0;
+
+    virtual std::pair<MetaTxsList, std::optional<AccountTxMarker>>
+    oldestAccountTxPageB(AccountTxPageOptions const& options) = 0;
+
+    virtual std::pair<MetaTxsList, std::optional<AccountTxMarker>>
+    newestAccountTxPageB(AccountTxPageOptions const& options) = 0;
+
+    virtual std::pair<
+        std::variant<
+            std::pair<std::shared_ptr<Transaction>, std::shared_ptr<TxMeta>>,
+            TxSearched>,
+        error_code_i>
+    loadTransaction(
+        uint256 const& id,
+        std::optional<ClosedInterval<uint32_t>> const& range,
+        error_code_i const& ec) = 0;
+
+    virtual bool
+    checkLedgerDBSpace(Config const& config) = 0;
+
+    virtual bool
+    checkTransactionDBSpace(Config const& config) = 0;
```

Comment:
It's not clear what is the meaning of the value returned by these functions. I think it would be better to change the name to `ledgerDBHasSpace()`. It's more obvious what the meaning of the returned value is.

Reply:
Done.